pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -15,13 +15,17 @@ const (
 	PrometheusNotAvailable         = "PrometheusNotAvailable"
 	PrometheusNotReconciled        = "PrometheusNotReconciled"
 	PrometheusDegraded             = "PrometheusDegraded"
+	ResourceSelectorIsNil          = "ResourceSelectorNil"
 	CannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
 	AvailableMessage               = "Monitoring Stack is available"
 	SuccessfullyReconciledMessage  = "Monitoring Stack is successfully reconciled"
+	ResourceSelectorIsNilMessage   = "No resources will be discovered, ResourceSelector is nil"
+	ResourceDiscoveryOnMessage     = "Resource discovery is operational"
 	NoReason                       = "None"
 )
 
-func updateConditions(msConditions []v1alpha1.Condition, prom monv1.Prometheus, generation int64, recError error) []v1alpha1.Condition {
```

Comment:
This is unfortunately necessary since the resource discovery condition is set based on a field of `MonitoringStack`. 
See https://github.com/rhobs/observability-operator/pull/223/files#diff-3c84e774a1e9ef2eafc525a973f4b4a273410b6b242fabda42e514c524328a38R64
And since the conditions are part of that anyway I simply changed the argument, instead of passing both the conditions list and the monitoring stack.

Reply:
Ahh yes, missed that detail!